### PR TITLE
(#202) Add Formula 5.10 from NEN-EN-1993.5:2008

### LINKED ISSUE #202: [✨ Feature request]: Add Formula 5.10 from NEN-EN-1993.5:2008
### Description

Add Formula 5.10 from NEN-EN-1993.5:2008

### Acceptance Criteria

- Formula 
- Latex
- Unit test
- Docstrings 

### Dependencies

_No response_

### Confirmation

- [X] I have checked that this feature does not already exist in the issue tracker.
- [X] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by Klaassicking**
## Description

Add Formula 5.10 from NEN-EN-1993.5:2008

$\rho = \left(2 \cdot \frac{V_{Ed}}{V_{pl,Rd}} - 1\right)^2$

Fixes #202

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by egarciamendez** [comment]
$\rho = \left(2 \cdot \frac{V_{Ed}}{V_{pl,Rd}} - 1\right)^2 = \left(2 \cdot \frac{100}{400} - 1\right)^2 = 0.25$

**Review by egarciamendez — APPROVED** [review]
ready to go 🚀

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+from blueprints.codes.eurocode.nen_en_1993_5_2008 import NEN_EN_1993_5_2008
+from blueprints.codes.formula import Formula
+from blueprints.codes.latex_formula import LatexFormula
+from blueprints.type_alias import DIMENSIONLESS, KN
+from blueprints.validations import raise_if_less_or_equal_to_zero
+
+
+class Form5Dot10ReductionFactorShear(Formula):
+    """Class representing formula 5.10 for reduction factor for shear resistance of the cross-section."""
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:11]: image

image

```suggestion
    """Class representing formula 5.10 for reduction factor for shear area."""
```

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+"""Formula 5.10 from NEN-EN 1993-5:2008 Chapter 5 - Ultimate limit state."""
+
+from blueprints.codes.eurocode.nen_en_1993_5_2008 import NEN_EN_1993_5_2008
+from blueprints.codes.formula import Formula
+from blueprints.codes.latex_formula import LatexFormula
+from blueprints.type_alias import DIMENSIONLESS, KN
+from blueprints.validations import raise_if_less_or_equal_to_zero
+
+
+class Form5Dot10ReductionFactorShear(Formula):
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:10]:
```suggestion
class Form5Dot10ReductionFactorShearArea(Formula):
```

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    label = "5.10"
+    source_document = NEN_EN_1993_5_2008
+
+    def __init__(
+        self,
+        v_ed: KN,
+        v_pl_rd: KN,
+    ) -> None:
+        """[:math:`ρ`] Calculate the reduction factor for shear resistance of the cross-section [-].
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:21]:
```suggestion
        """[:math:`ρ`] Calculate the reduction factor for shear area of the cross-section [-].
```

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        super().__init__()
+        self.v_ed: float = v_ed
+        self.v_pl_rd: float = v_pl_rd
+
+    @staticmethod
+    def _evaluate(
+        v_ed: KN,
+        v_pl_rd: KN,
+    ) -> DIMENSIONLESS:
+        """Evaluates the formula for reduction factor for shear resistance."""
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:41]:
```suggestion
        """Evaluates the formula for reduction factor for shear area."""
```

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        raise_if_less_or_equal_to_zero(v_ed=v_ed, v_pl_rd=v_pl_rd)
+        return (2 * v_ed / v_pl_rd - 1) ** 2
+
+    def latex(self) -> LatexFormula:
+        """Returns LatexFormula object for formula 5.10."""
+        return LatexFormula(
+            return_symbol=r"\rho",
+            result=str(self),
+            equation=r"\left(2 \cdot \frac{V_{Ed}}{V_{pl,Rd}} - 1\right)^2",
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:50]: why not use the latex_fraction function here?

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            numeric_equation=rf"\left(2 \cdot \frac{{{self.v_ed}}}{{{self.v_pl_rd}}} - 1\right)^2",
```
Comment (SZeltaat) [inline blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:51]: why not use the latex_fraction function here?

**Review by SZeltaat — COMMENTED** [review]
Good job and thanks for contributing! 
I have some comments on the name. rho is a reduction factor on the shear area and not necessarily shear resistance. It could indeed affect the shear resistance, but especially in this case it is not used for that purpose. So I believe the name should change.

ALSO DONT FORGET to add to the docs in **docs\source\codes\eurocode**. There are no md files there yet. So they should be created first.

Good luck!

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py (hunk 1 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:11]: That sounds indeed like a better description. 👍

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py (hunk 5 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:50]: Sharp observation! There isn't a specific reason, but for the sake of simplicity, let's leave this one as it is.

File: blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py (hunk 6 above)
Comment (egarciamendez) [reply blueprints/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/formula_5_10.py:51]: Sharp observation! There isn't a specific reason, but for the sake of simplicity, let's leave this one as it is.

**Comment by egarciamendez** [comment]
> Good job and thanks for contributing!
> 
>     * [x]  I have some comments on the name. rho is a reduction factor on the shear area and not necessarily shear resistance. It could indeed affect the shear resistance, but especially in this case it is not used for that purpose. So I believe the name should change.
> 
>     * [ ]  ALSO DONT FORGET to add to the docs in **docs\source\codes\eurocode**. There are no md files there yet. So they should be created first.
> 
> 
> Good luck!

Docs must be added indeed. Let's add that action to the backlog 👍

**Pull request merged**